pkg/kubelet/cm/helpers_linux.go

```diff
@@ -20,39 +20,84 @@ import (
-}
-
 // GetCgroupSubsystems returns information about the mounted cgroup subsystems
-func getCgroupSubsystems() (*cgroupSubsystems, error) {
-	// Get all cgroup mounts.
+func GetCgroupSubsystems() (*CgroupSubsystems, error) {
+	// get all cgroup mounts.
 	allCgroups, err := libcontainercgroups.GetCgroupMounts()
 	if err != nil {
-		return &cgroupSubsystems{}, err
+		return &CgroupSubsystems{}, err
 	}
 	if len(allCgroups) == 0 {
-		return &cgroupSubsystems{}, fmt.Errorf("failed to find cgroup mounts")
+		return &CgroupSubsystems{}, fmt.Errorf("failed to find cgroup mounts")
 	}
 
-	//TODO(@dubstack) should we trim to only the supported ones
 	mountPoints := make(map[string]string, len(allCgroups))
 	for _, mount := range allCgroups {
 		for _, subsystem := range mount.Subsystems {
 			mountPoints[subsystem] = mount.Mountpoint
 		}
 	}
-	return &cgroupSubsystems{
-		mounts:      allCgroups,
-		mountPoints: mountPoints,
+	return &CgroupSubsystems{
+		Mounts:      allCgroups,
+		MountPoints: mountPoints,
 	}, nil
 }
+
+// GetPodResourceRequest returns the pod requests for the supported resources.
+// Pod request is the summation of resource requests of all containers in the pod.
+func GetPodResourceRequests(pod *api.Pod) api.ResourceList {
+	requests := api.ResourceList{}
+	zeroQuantity := resource.MustParse("0")
+	for _, resource := range []api.ResourceName{api.ResourceCPU, api.ResourceMemory} {
+		for _, container := range pod.Spec.Containers {
+			quantity := container.Resources.Requests[resource]
+			if quantity.Cmp(zeroQuantity) == 1 {
+				delta := quantity.Copy()
+				if _, exists := requests[resource]; !exists {
+					requests[resource] = *delta
+				} else {
+					delta.Add(requests[resource])
+					requests[resource] = *delta
+				}
+			}
+		}
+	}
+	return requests
+}
+
+// GetPodResourceLimits returns the pod limits for the supported resources
+// Pod limit is the summation of resource limits of all containers
+// in the pod. If limit for a particular resource is not specified for
+// even a single container then we return the node resource capacity
+// as the pod limit for the particular resource.
+func GetPodResourceLimits(pod *api.Pod, nodeInfo *api.Node) api.ResourceList {
+	capacity := nodeInfo.Status.Capacity
+	limits := api.ResourceList{}
+	zeroQuantity := resource.MustParse("0")
+	for _, resource := range []api.ResourceName{api.ResourceCPU, api.ResourceMemory} {
+		for _, container := range pod.Spec.Containers {
+			quantity := container.Resources.Limits[resource]
+			if quantity.Cmp(zeroQuantity) == 1 {
+				delta := quantity.Copy()
+				if _, exists := limits[resource]; !exists {
+					limits[resource] = *delta
+				} else {
+					delta.Add(limits[resource])
+					limits[resource] = *delta
+				}
+			} else {
+				// if limit not specified for a particular resource in a container
+				// we default the pod resource limit to the resource capacity of the node
+				if cap, exists := capacity[resource]; exists {
```

Comment:
does this mean we will show limits whose reported value is greater than node capacity if a pod has 2 containers and ctr1 has limits, and ctr2 does not?